rust/auction/program/src/instruction.rs

```diff
@@ -30,10 +31,20 @@ pub enum AuctionInstruction {
     /// Create a new auction account bound to a resource, initially in a pending state.
     ///   0. `[signer]` The account creating the auction, which is authorised to make changes.
     ///   1. `[writable]` Uninitialized auction account.
-    ///   2. `[]` Rent sysvar
-    ///   3. `[]` System account
+    ///   2. `[writable]` Auction extended data account.
+    ///   3. `[]` Rent sysvar
+    ///   4. `[]` System account
     CreateAuction(CreateAuctionArgs),
 
+    /// Create a new auction account bound to a resource, initially in a pending state.
+    /// The only one difference with above instruction it's instant_sale_price parameter in CreateAuctionArgsV2
+    ///   0. `[signer]` The account creating the auction, which is authorised to make changes.
+    ///   1. `[writable]` Uninitialized auction account.
```

Comment:
By placing this here instead of at the end of the enum, you have now offset the contract commands for every command after it and supplanted the contract command that used to have this number(ClaimBid). This means when old UIs call this contract expecting to hit ClaimBid, they'll hit CreateAuctionV2 instead, and when they call anything else, they'll always be "off by one." You need to place any new contract commands at the end of the Instruction enum so as not to disrupt older clients, because the clients pass u8s to tell the program which command to use and these are just indexing into the Instruction enum.